pymc/sampling/forward.py

```diff
@@ -497,22 +499,148 @@ def sample_posterior_predictive(
-    Thin a sampled inferencedata by keeping 1 out of every 5 draws
-    before passing it to sample_posterior_predictive
+    The behavior of ``var_names``
+    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+    The function returns posterior predictive samples of any variable included in ``var_names``,
+    conditioned on the values of other random variables found in the trace.
+    To ensure the samples are internally consistent, any random variable that depends
+    on another random variable that is being sampled is itself sampled.
+    The final list of variables being sampled is shown in the log output.
+
+    Note that if a random variable has no dependency on other random variables,
+    posterior predictive sampling is equivalent to prior sampling.
+    Likewise, if all random variables are being sampled, the behavior is equivalent to
+    that of :func:`~pymc.sample_prior_predictive`.
+
+    .. warning:: A random variable included in ``var_names`` will never be copied from the posterior group. It will always be sampled as described above. If you want, you can copy manually via ``idata.posterior_predictive["var_name"] = idata.posterior["var_name"]``.
+
+
+    The following code block explores how the behavior changes with different var_names:
+
+    .. code:: python
+
+        from logging import getLogger
+        import pymc as pm
+
+        # Some environments like google colab suppress
+        # the default logging output of PyMC
+        getLogger("pymc").setLevel("INFO")
+
+        kwargs = {"progressbar": False, "random_seed": 0}
+
+        with pm.Model() as model:
+          x = pm.Normal("x")
+          y = pm.Normal("y")
+          z = pm.Normal("z", x + y**2)
+          det = pm.Deterministic("det", pm.math.exp(z))
+          obs = pm.Normal("obs", det, 1, observed=[20])
+
+          idata = pm.sample(tune=10, draws=10, chains=2, **kwargs)
+
+    Default behavior. Posterior predictive draws of ``obs``,
+    conditioned on the posterior draws of ``z`` found in the trace.
+
+    .. code:: python
+
+          pm.sample_posterior_predictive(idata, var_names=["obs"], **kwargs)
+          # Sampling: [obs]
+
+    Re-compute the deterministic variable ``det``,
+    conditioned on the posterior draws of ``z`` found in the trace.
+
+    .. code :: python
+
+          pm.sample_posterior_predictive(idata, var_names=["det"], **kwargs)
+          # Sampling: []
+
+    Posterior predictive draws of ``z`` and ``det``,
+    conditioned on the posterior draws of ``x``` and ``y``` found in the trace.
+
+    Note: It does not matter that ``z``` was not an observed variable.
+
+    .. code :: python
+
+          pm.sample_posterior_predictive(idata, var_names=["z", "det"], **kwargs)
+          # Sampling: [z]
+
+
+    Posterior predictive draws of ``y``, ``z`` and ``det``,
+    conditioned on the posterior draws of ``x`` found in the trace.
+
+    Note: The samples of ``y`` are equivalent to its prior, since it does not depend on any other variables.
```

Comment:
```suggestion
    Note: The samples of ``y`` are equivalent to its prior, since it does not depend on any other variables. That also means that the samples of ``z`` and ``det`` are a mixture of posterior samples of ``x`` and prior samples of ``y``, since ``z`` and ``det`` are a function of ``x`` and ``y``.
```